### PR TITLE
Implement Entity Component in React + TS

### PR DESCRIPTION
**Pull request by harsh3dev**
This PR solves issue #54 
This is how the project structure looks now:
image

### Overview
- The `EntityExampleComponent.tsx` is a temporary example component that displays the card and can be used as an example on how to use this component.
- This `EntityExampleComponent.tsx` shall be deleted later when the routing of the frontend is setup using [`React-Router`](https://reactrouter.com/en/main/start/overview).

#### Card Component (`src/components/Card.tsx`)
- The main component is `src/components/Card.tsx` which accepts few `props`.
- It displays information relevant to the props passed so that each page renders relevant details.
- For tooltip I have used `react-tooltip` package.
- The types and constants are exported from separate files.

- Also added `dark mode` and `light mode` functionality
- Using variables for colors in Tailwind CSS for consistency in styling.

Refer to this video for a quick look at the output:
https://www.loom.com/share/2e5be837be864bfcb5d02dff9ba5af17?sid=e87e552f-3ac4-4714-91da-04c4b5772d2c

### CODE REVIEW

File: frontend/package.json
Comment (arkid15r) [inline frontend/package.json:1]: I think it's fine to keep the non-dependencies attributes unchanged here.

File: frontend/index.html [hunk 1]
```diff
@@ -1,8 +1,8 @@
-<!DOCTYPE html>
+<!doctype html>
 <html lang="en">
   <head>
     <meta charset="UTF-8" />
-    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.0.0-beta3/css/all.min.css" />
```
Comment (arkid15r) [inline frontend/index.html]: Shouldn't we use `@fortawesome/react-fontawesome` instead?

File: frontend/src/lib/constants.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    level: "Production",
+  },
+  flagship: {
+    color: "#38a047",
+    icon: " text-white fa-solid fa-flag",
+    level: "Flagship",
+  },
+};
+
+export const urlMappings = [
```
Comment (arkid15r) [inline frontend/src/lib/constants.ts]: It looks like `data.tsx` already contains same mapping.

File: frontend/src/lib/types.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    idx_forks_count: number;
+    idx_leaders: string[];
+    idx_level: string;
+    idx_name: string;
+    idx_stars_count: number;
+    idx_summary: string;
+    idx_topics: string[];
+    idx_type: string;
+    idx_updated_at: number; // UNIX timestamp
+    idx_url: string;
```
Comment (arkid15r) [inline frontend/src/lib/types.ts:23]: I'd prefer to get rid of `idx_` prefix in interfaces if possible. It'd be more readable.
Do you think it's something that can be done?

**Review by arkid15r — COMMENTED** [review]
@harsh3dev amazing work!
This PR gets us closer to the React-based frontend milestone.

Please find my questions/comments below:

File: frontend/index.html (hunk 1 above)
Comment (harsh3dev) [reply frontend/index.html]: Yes, we can use react-fontawesome and i will replace it.

File: frontend/src/lib/constants.ts (hunk 2 above)
Comment (harsh3dev) [reply frontend/src/lib/constants.ts]: Okay, I see there are these redundant constants, I must have to clean them.

File: frontend/src/lib/types.ts (hunk 3 above)
Comment (harsh3dev) [reply frontend/src/lib/types.ts:23]: Sure, I can remove the `idx_` prefix and make it more readable

**Comment by harsh3dev** [comment]
@arkid15r I went through the suggestions and comments and I will update the commit soon.
Thanks

**Comment by harsh3dev** [comment]
@arkid15r @tsu-ki Please have a look at the changes. Waiting for your approval.
Thanks.

**Review by arkid15r — APPROVED** [review]
Let's merge it!
Great work @harsh3dev 👍

**Pull request merged**